blank/webpack.config.js

```diff
@@ -99,192 +120,188 @@ const stylesLoaderConfig = {
       }),
       new webpack.HotModuleReplacementPlugin(),
       imageminPlugin,
-      new webpack.EnvironmentPlugin({
-        HUBTYPE_API_URL: null,
-        BOTONIC_TARGET: BOTONIC_TARGETS.DEV,
+      new webpack.DefinePlugin({
+        IS_BROWSER: true,
+        HUBTYPE_API_URL: JSON.stringify(process.env.HUBTYPE_API_URL),
       }),
     ],
   }
 }
 
 function botonicWebchatConfig(mode) {
   return {
-    optimization: {
-      minimizer: [terserPlugin],
-    },
+    optimization: optimizationConfig,
     mode: mode,
     devtool: sourceMap(mode),
     target: 'web',
-    entry: path.resolve('webpack-entries', 'webchat-entry.js'),
+    entry: path.resolve(WEBPACK_ENTRIES.DIR, WEBPACK_ENTRIES.WEBCHAT),
     module: {
-      rules: [babelLoaderConfig, fileLoaderConfig, stylesLoaderConfig],
+      rules: [
+        babelLoaderConfig,
+        fileLoaderConfig('assets'),
+        stylesLoaderConfig,
+      ],
     },
     output: {
-      path: path.resolve(__dirname, 'dist'),
-      filename: 'webchat.botonic.js',
-      library: 'Botonic',
-      libraryTarget: 'umd',
+      filename: WEBCHAT_FILENAME,
+      library: BOTONIC_LIBRARY_NAME,
+      libraryTarget: UMD_LIBRARY_TARGET,
       libraryExport: 'app',
-      publicPath: './',
+      path: OUTPUT_PATH,
     },
     resolve: resolveConfig,
     plugins: [
       new HtmlWebpackPlugin({
-        template: path.resolve(botonicPath, 'src', 'webchat.template.html'),
+        template: path.resolve(BOTONIC_PATH, 'src', TEMPLATES.WEBCHAT),
         filename: 'index.html',
       }),
       imageminPlugin,
-      new webpack.EnvironmentPlugin({
-        HUBTYPE_API_URL: null,
-        WEBCHAT_PUSHER_KEY: null,
-        BOTONIC_TARGET: 'webchat',
+      new webpack.DefinePlugin({
+        IS_BROWSER: true,
+        HUBTYPE_API_URL: JSON.stringify(process.env.HUBTYPE_API_URL),
+        WEBCHAT_PUSHER_KEY: JSON.stringify(process.env.WEBCHAT_PUSHER_KEY),
       }),
     ],
   }
 }
 
 function botonicWebviewsConfig(mode) {
   return {
-    optimization: {
-      minimizer: [terserPlugin],
-    },
+    optimization: optimizationConfig,
     mode: mode,
     devtool: sourceMap(mode),
     target: 'web',
-    entry: path.resolve('webpack-entries', 'webviews-entry.js'),
+    entry: path.resolve(WEBPACK_ENTRIES.DIR, WEBPACK_ENTRIES.WEBVIEWS),
     output: {
-      path: path.resolve(__dirname, 'dist/webviews'),
       filename: 'webviews.js',
       library: 'BotonicWebview',
-      libraryTarget: 'umd',
+      libraryTarget: UMD_LIBRARY_TARGET,
       libraryExport: 'app',
+      path: path.resolve(OUTPUT_PATH, 'webviews'),
     },
     module: {
       rules: [
         babelLoaderConfig,
-        {
-          test: /\.(png|svg|jpg|gif)$/,
-          use: [
-            {
-              loader: 'file-loader',
-              options: {
-                outputPath: '../assets',
-              },
-            },
-          ],
-        },
+        fileLoaderConfig('../assets'),
```

Comment:
Why here is in a previous folder (`../`)?

Reply:
this is because when generating the `dist` folder, we are telling `webviews` dir, to source `assets` from a previous directory.
Ex:
```
- /assets
- /webviews
- webchat.botonic.js
- etc.
 ```